addons/sale_timesheet/models/hr_timesheet.py

```diff
@@ -200,3 +200,26 @@ def _timesheet_convert_sol_uom(self, sol, to_unit):
 
     def _is_updatable_timesheet(self):
         return super()._is_updatable_timesheet and self._is_not_billed()
+
+    def _timesheet_preprocess_get_accounts(self, vals):
+        so_line = self.env['sale.order.line'].browse(vals.get('so_line'))
+        if not so_line:
+            return super()._timesheet_preprocess_get_accounts(vals)
+        distribution = so_line.analytic_distribution
+        if not distribution:
+            return super()._timesheet_preprocess_get_accounts(vals)
```

Comment:
Equivalent to:

```suggestion
        if not so_line or not (distribution := so_line.analytic_distribution):
            return super()._timesheet_preprocess_get_accounts(vals)
```

or 

```suggestion
        if not (so_line and (distribution := so_line.analytic_distribution):
            return super()._timesheet_preprocess_get_accounts(vals)
```